### PR TITLE
[CR] Add stimulant/painkiller overdose symptoms.

### LINKED ISSUE #1291: Early onset symptoms for drug OD.
Currently you have 0 warning before a death from drug overdose occurs, which as has been pointed out in http://www.cataclysmdda.com/smf/index.php?topic=390.30 this is both unrealistic and confusing.
Add various early-onset indicators and symptoms for various drugs, not so much to get people to back off, as it's often too late, but for a less confusing experience and possibly to educate players about the dangers of drugs with near-misses.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1366882-early-onset-symptoms-for-drug-od?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SpencerMichaels**
This resolves #1291 by adding overt symptoms for both painkiller and stimulant overdose. The symptoms are split into four levels for each, and they stack with one another (i.e. low intensity symptoms will keep occurring when you "graduate" to higher ones ). I'm not sure what good values for some of these status effects are, so I'll consider this in need of review for now.

Sources are these two pages: [opioids](http://harmreduction.org/issues/overdose-prevention/overview/overdose-basics/recognizing-opioid-overdose/), [stimulants](http://harmreduction.org/issues/overdose-prevention/overview/stimulant-overamping-basics/recognizing-stimulant-overamping/).

### Stimulant overdose (250+ kills)
1. `stim > 75`: nausea
2. `stim > 110`: shakes
3. `stim > 170`: shortness of breath (winded)
4. `stim > 210`: muscle spasms (1/20 chance per turn of being downed)

### Painkiller overdose (-200 stim or +240 painkillers kills)
1. `stim < -60 || pkill > 130`: fatigue (-stim/10 fatigue mod every 10 mins — is this too low/high?)
2. `stim < 110 || pkill > 145`: dizziness (1/15 change of moves -= 10-30, if that occurs, also 1/3 chance of stumbling, i.e. gain "downed" for 3-10 turns) 
3. `stim < 170 || pkill > 160`: shortness of breath (winded)
4. `stim < 210 || pkill > 200`: blackouts (1/30 chance of downed + blind + 300-600 turns of unconsciousness)

### CODE REVIEW

**Comment by AMurkin** [comment]
Both links lead to the same page.

**Comment by sxmichaels** [comment]
Oops, sorry. Fixed now.

**Comment by Night-Pryanik** [comment]
> stim > 75: nausea
 stim > 170: shortness of breath (winded)
 stim > 210: muscle spasms (1/20 chance per turn of being downed)

But the code tells that with stim > 75 character start feeling cold, with stim > 170 character start seeing visuals, and with stim > 210 character begin hallucinating.

**Comment by sxmichaels** [comment]
…aaand that would be me editing an update the wrong branch. That was an old version. Apparently I need to pay closer attention to my PRs. :/  Fixed now, sorry again.

**Comment by AMurkin** [comment]
`moves` changes should be done by `mod_moves()`.

File: src/player.cpp [hunk 1]
```diff
@@ -6142,6 +6142,65 @@ void player::suffer()
...
+    }
+    if ( stim > 170 ) {
+        if ( !has_effect( effect_winded ) && calendar::once_every( MINUTES(5) ) ) {
+            add_msg(m_bad, _("You feel short of breath.") );
+            add_effect( effect_winded, MINUTES(5) + 1 );
+        }
+    }
+    if ( stim > 110 ) {
+        if ( !has_effect( effect_shakes ) && calendar::once_every( MINUTES(5) ) ) {
+            add_msg( _("You start to shake uncontrollably.") );
```
Comment (Night-Pryanik) [inline src/player.cpp]: This message will reappear every 5 minutes as long as character's `stim` > 110, right? Maybe then remove the "start" part from it and leave simple "You shake uncontrollably."?
Also, I think that message reappearing every 5 minutes will easily spam the entire message log.

File: src/player.cpp [hunk 2]
```diff
@@ -6142,6 +6142,65 @@ void player::suffer()
...
+            if ( one_in(3) && !has_effect( effect_downed ) ) {
+                add_msg_if_player(m_bad, _("You stumble and fall over!"));
+                add_effect( effect_downed, rng(3, 10) );
+            }
+        }
+    }
+    if ( stim < -60 || pkill > 130 ) {
+        if( calendar::once_every( MINUTES( 10 ) ) ) {
+            add_msg(m_warning, _("You feel tired...") );
+            mod_fatigue( stim / (-10) );
```
Comment (Night-Pryanik) [inline src/player.cpp]: With positive `stim` this will lower the character's fatigue.

**Comment by Night-Pryanik** [comment]
Astyling would be great.

File: src/player.cpp (hunk 1 above)
Comment (sxmichaels) [reply src/player.cpp]: Yeah, I was concerned about message spam to some extent. How far apart should the messages be spaced, then? Is 10 minutes sufficient?

**Comment by sxmichaels** [comment]
Thanks for the feedback. Updated and astyled.

File: src/player.cpp [hunk 3]
```diff
@@ -6142,6 +6142,67 @@ void player::suffer()
     if (has_artifact_with(AEP_ATTENTION)) {
         add_effect( effect_attention, 3 );
     }
+
+    // Stim +250 kills
+    if ( stim > 210 ) {
+        if ( one_in( 20 ) && !has_effect( effect_downed ) ) {
+            add_msg_if_player(m_bad, _("Your muscles spasm, and you fall to the ground!"));
```
Comment (kevingranade) [inline src/player.cpp]: Maybe if the player is downed print "Your muscles spasm!"

File: src/player.cpp (hunk 1 above)
Comment (Night-Pryanik) [reply src/player.cpp]: Yeah, at least 10 minutes is good imo.

**Comment by Night-Pryanik** [comment]
I fear that you didn't pushed your changes for astyling.

**Comment by sxmichaels** [comment]
@Night-Pryanik By astyling you mean running `make astyle`, right? I did that before the last commit, and doing so again doesn't produce any changes.

**Comment by Night-Pryanik** [comment]
Personally I use the latest downloaded version of astyle (from http://astyle.sourceforge.net/) and applying the `.astylerc` config.

**Comment by kevingranade** [comment]
>  fatigue (-stim/10 fatigue mod every 10 mins — is this too low/high?)

Wow I thought this was ok, but it is... not.  Once the fatigue thing kicks in it accumulates nearly as fast as you can sleep it off.  It's pretty cripling.  Probably back all the way off to 1 or 2 per tick.

**Comment by sxmichaels** [comment]
Here's a tweaked version. Fatigue is now `rng(1, 2)` every 10 minutes.

**Pull request merged**